release-24.3: Make uuid parsing errors as safe string

Abhinav1299 opened the pull request:
Backport:
  * 1/1 commits from "pkg/util/uuid: Make uuid parsing errors as safe string" (#147762)
  * 1/1 commits from "pkg/util/uuid: simplify error message in UUID UnmarshalBinary method" (#149190)

Please see individual PRs for details.

/cc @cockroachdb/release

----
This patch helps to make the uuid parsing errors as safe so that when the roachpb.Key is
logged, the error message is not redacted.

The prior implementation of uuid errors followed the fmt.Errorf strings which are
categorised as unsafe by default when logged using %v verb of string formatting.
This patch convert those error objects to use redact.Safe implementation.

----

Release justification: Fixing the high priority error log line that was over redacted.

Abhinav1299 committed:
pkg/util/uuid: Make uuid parsing errors as safe string

This patch helps to make the uuid parsing errors as safe so that when the
roachpb.Key is logged, the error message is not redacted.

The prior implementation of uuid errors followed the fmt.Errorf strings which
are categorized as unsafe by default when logged using %v verb of string formatting.
This patch convert those error objects to use redact.Safe implementation.

Epic: CRDB-37533
Part of: CRDB-44885
Release note: None

Abhinav1299 committed:
pkg/util/uuid: simplify error message in UUID UnmarshalBinary method

Updated the error message in the UnmarshalBinary method of the UUID struct to directly use the length of the data instead of a redaction function.

Epic: CRDB-37533
Part of: CRDB-44885
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/149224)
<!-- Reviewable:end -->

Abhinav1299 committed:
pkg/util/uuid: add redact dependency for improved error handling

This update introduces a new dependency on the `redact` package from CockroachDB to enhance error handling in the UUID library. This change allows for better management of sensitive information in error messages.

Epic: CRDB-37533
Part of: CRDB-44885
Release note: None
